### PR TITLE
Implement TouchArea::scroll-event

### LINKED ISSUE #1280: WheelEvents callback
TouchArea should support wheel event.

Options:
 1. add a new `callback wheel-event(Orientation, delta)` (what type should the delta be?)
 
 2. add things in the current pointer-event.  `PointerEventKind.vertical-wheel` / `horizontal-wheel` or should it be in PointerEventButton ?     `PointerEvent.wheel-delta` ?

### PR DESCRIPTION
**Pull request by ogoffart**
Closes #1280

### CODE REVIEW

**Comment by ogoffart** [comment]
Thinking of it, another possible name would be `scrolled`  analog to `clicked` and `moved`

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ All notable changes to this project are documented in this file.
  - Added `Number`, `Decimal` variant to enum `InputType`
  - Added `spacing-horizontal` and `spacing-vertical` in `GridLayout`
  - Fixed conversion in array of array of structs (#3574)
+ - Added `scroll-event`` callback to `TouchArea`
```
Comment (tronical) [inline CHANGELOG.md:18]:
```suggestion
 - Added `scroll-event` callback to `TouchArea`
```

File: internal/core/items/flickable.rs [hunk 2]
```diff
@@ -251,9 +251,7 @@ impl FlickableData {
                     InputEventFilterResult::ForwardEvent
                 }
             }
-            MouseEvent::Wheel { position, .. } => {
-                InputEventFilterResult::InterceptAndDispatch(MouseEvent::Moved { position })
-            }
+            MouseEvent::Wheel { .. } => InputEventFilterResult::ForwardEvent,
```
Comment (tronical) [inline internal/core/items/flickable.rs:254]: I think this means that the docs should be changed for the flickable "algorithm" as well. Currently it says:

> 2. Any wheel events are interpreted as attempt of the user to scroll.

I think that section can simply be removed? (And the others renumbered)

**Review by tronical — APPROVED** [review]

**Comment by tronical** [comment]
> Thinking of it, another possible name would be `scrolled` analog to `clicked` and `moved`

For the callback? Hmm, but that sounds like it would eliminate the possibility of ignoring the event. We use the accept/ignore pattern currently for *-event names, don't we?

**Review by tronical — APPROVED** [review]

**Pull request merged**